### PR TITLE
Fix RPCViolation R3019 Top level property names should not be repeate…

### PR DESCRIPTION
**Pull request by naveedaz**
…d inside the properties bag for ARM resource

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json [hunk 1]
```diff
@@ -15651,12 +15624,7 @@
         "properties": {
           "description": "FunctionEnvelope resource specific properties",
           "properties": {
-            "name": {
-              "description": "Function name.",
-              "type": "string",
-              "readOnly": true
-            },
-            "functionAppId": {
+            "function_app_id": {
```
Comment (marstr) [inline specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json]: Was this a violation? I thought we preferred the other naming convention.

**Comment by marstr** [comment]
ARM folks (@ravbhatnagar or @simongdavies), mind taking a look and ensuring that the removed names are now in compliance with your expectations? Monday, I'll still need to generate some of the SDKs to ensure nothing is broken.

File: specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json (hunk 1 above)
Comment (naveedaz) [reply specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json]: This wasn't a violation. It snuck in due to a merge issue with the generated swaggers. Fixed it.

**Comment by marstr** [comment]
We made some changes a few weeks ago, included was moving back to `master` as the default branch. For now `current` and `master` are setup to advance together. I've updated this PR to target `master` to conform to the new policy.

**Comment by naveedaz** [comment]
@marstr Thanks. So future checkins should be against master branch?

@ravbhatnagar or @simongdavies. Please help review.

**Comment by ravbhatnagar** [comment]
@naveedaz - Since these are breaking changes, lets add them to a newer api-version. Other than that looks good.

**Comment by marstr** [comment]
> @naveedaz says:
So future checkins should be against master branch?

Yep!

> @ravbhatnagar says:
Since these are breaking changes, lets add them to a newer api-version.

I believe that means that this PR should be closed, or at very least changed to be seeking surpressions instead of fixing the underlying issues. @naveedaz, does that work for you?

**Comment by naveedaz** [comment]
The REST APIs do not already use the properties that were removed. They were remnants from the RDFE data models that we re-purposed for ARM, when ARM was first introduced. Exposing a new version of APIs that simply have no functional change on the service end is not really helpful and will only delay getting these errors fixed. The downlevel SDKs and clients generated will still be a different version anyhow. We want to get to 0 errors quickly. There will be another round of fixes for these RPC violations soon and we don't want to keep spinning up new versions on the service side for that unless it breaks service functionality.

**Comment by marstr** [comment]
Thanks for getting back promptly, @naveedaz. There are some important points to talk about here.

> @naveedaz says:
We want to get to 0 errors quickly. There will be another round of fixes for these RPC violations soon and we don't want to keep spinning up new versions on the service side for that unless it breaks service functionality.

We appreciate that immensely! I believe we all want to do that in the way that causes the least customer pain. Certainly, adding a lot of server-side versions is undesirable too. I think it's safe for me to speak for @ravbhatnagar here, and say that we would both recommend including these name changes and your other upcoming changes into a single new API Version, instead of having two.

> @naveedaz says:
The downlevel SDKs and clients generated will still be a different version anyhow.

This is true for some languages, but we're trying to move away from this approach. We'd like to a have world where SDKs support all API Versions in a single SDK version. Disentangling the two versions will result in fewer breaking changes that customers have to deal with. The Go, Python*, and Ruby SDKs have already moved to an approach where a new API Version is a minor version bump, instead of a major.

* - Python has done this for the larges, most prone to breaking services and left the others alone.

(Mentioning @devigned for visibility)

**Comment by devigned** [comment]
I agree with @marstr and @ravbhatnagar. It would be preferred to leave existing API versions as they are and not introduce a breaking change to those published packages. It would be best to make **all** of the changes in a new API version.

Going back in time to fix old API versions is painful for customers who have taken dependencies on this code.

**Comment by ravbhatnagar** [comment]
had a chat with Navy about this RPC validation error which is being addressed in this PR. Its a tough one since adding to all existing models in Web RP is not straightforward and it will be quite a bit of work. We discussed multiple options on this. I will follow up to see which option we would like to go with.

**Comment by ravbhatnagar** [comment]
Since the ARM platform capability which would have been impacted by violating this rule has been de-prioritized, we can suppress this rule for Web RP since fixing this requires a major overhaul at their end. considering they have close to 300 operations and a lot are impacted by this, we are fine letting Web suppress this. There are other reasons for which the rule should still apply, but considering the amount of time Web will have to spend to fix this and the benefit we will get out of it, it seems reasonable to let them bypass. 
@salameer @veronicagg - thoughts?

**Comment by naveedaz** [comment]
Closing out this PR. Will send out a suppression PR.

**Pull request closed without merging**